components/playground/playground.go

```diff
@@ -785,68 +786,82 @@ func (p *Playground) bootCluster(ctx context.Context, env *environment.Environme
 	p.booted = true
 
 	var succ []string
-	for _, db := range p.tidbs {
-		prefix := color.YellowString("Waiting for tidb %s ready ", db.Addr())
-		bar := progress.NewSingleBar(prefix)
-		bar.StartRenderLoop()
-		if s := checkDB(db.Addr()); s {
-			succ = append(succ, db.Addr())
-			bar.UpdateDisplay(&progress.DisplayProps{
-				Prefix: prefix,
-				Mode:   progress.ModeDone,
-			})
-		} else {
-			bar.UpdateDisplay(&progress.DisplayProps{
-				Prefix: prefix,
-				Mode:   progress.ModeError,
-			})
+	if len(p.tidbs) > 0 {
+		var wg sync.WaitGroup
+		bars := progress.NewMultiBar(color.YellowString("Waiting for tidb instances ready\n"))
+		for _, db := range p.tidbs {
+			wg.Add(1)
+			prefix := color.YellowString(db.Addr())
+			bar := bars.AddBar(prefix)
+			go func(dbInst *instance.TiDBInstance) {
+				defer wg.Done()
+				if s := checkDB(dbInst.Addr(), options.tidb.UpTimeout); s {
+					succ = append(succ, dbInst.Addr())
```

Comment:
may lose some successfully started instances, because slice is not thread safe. it's better to use channel so.

Reply:
Oh, yes. It's better to fix that...